contracts/DefaultStrategy.sol

```diff
@@ -0,0 +1,517 @@
+    }
+
+    /**
+     * @notice Calculates how many votes should be withdrawn from each
+     * deprecated group.
+     * @param withdrawal The total amount of votes that needs to be withdrawn.
+     * @return deprecatedGroupsWithdrawn The array of deprecated groups to be
+     * withdrawn from.
+     * @return deprecatedWithdrawalsPerGroup The amount of votes to withdraw
+     * from the respective deprecated group in `deprecatedGroupsWithdrawn`.
+     * @return numberDeprecatedGroupsWithdrawn The number of groups in
+     * `deprecatedGroupsWithdrawn` that have a non zero withdrawal.
+     * @return remainingWithdrawal The number of votes that still need to be
+     * withdrawn after withdrawing from deprecated groups.
+     * @dev Non zero entries of `deprecatedWithdrawalsPerGroup` will be exactly
+     * a prefix of length `numberDeprecatedGroupsWithdrawn`.
+     */
+    function getDeprecatedGroupsWithdrawalDistribution(uint256 withdrawal)
+        internal
+        returns (
+            address[] memory deprecatedGroupsWithdrawn,
+            uint256[] memory deprecatedWithdrawalsPerGroup,
+            uint256 numberDeprecatedGroupsWithdrawn,
+            uint256 remainingWithdrawal
+        )
+    {
+        remainingWithdrawal = withdrawal;
+        uint256 numberDeprecatedGroups = IManager(manager).getDeprecatedGroupsLength();
+        deprecatedGroupsWithdrawn = new address[](numberDeprecatedGroups);
+        deprecatedWithdrawalsPerGroup = new uint256[](numberDeprecatedGroups);
+        numberDeprecatedGroupsWithdrawn = 0;
+
+        for (uint256 i = 0; i < numberDeprecatedGroups; i++) {
+            numberDeprecatedGroupsWithdrawn++;
+            deprecatedGroupsWithdrawn[i] = IManager(manager).getDeprecatedGroup(i);
+            uint256 currentVotes = account.getCeloForGroup(deprecatedGroupsWithdrawn[i]);
+            deprecatedWithdrawalsPerGroup[i] = Math.min(remainingWithdrawal, currentVotes);
+            remainingWithdrawal -= deprecatedWithdrawalsPerGroup[i];
+
+            if (currentVotes == deprecatedWithdrawalsPerGroup[i]) {
+                if (!IManager(manager).removeDeprecatedGroup(deprecatedGroupsWithdrawn[i])) {
+                    revert FailedToRemoveDeprecatedGroup(deprecatedGroupsWithdrawn[i]);
+                }
+                emit GroupRemoved(deprecatedGroupsWithdrawn[i]);
+            }
+
+            if (remainingWithdrawal == 0) {
+                break;
+            }
+        }
+
+        return (
+            deprecatedGroupsWithdrawn,
+            deprecatedWithdrawalsPerGroup,
+            numberDeprecatedGroupsWithdrawn,
+            remainingWithdrawal
+        );
+    }
+
+    /**
+     * @notice Returns the storage, major, minor, and patch version of the contract.
+     * @return Storage version of the contract.
+     * @return Major version of the contract.
+     * @return Minor version of the contract.
+     * @return Patch version of the contract.
+     */
+    function getVersionNumber()
```

Comment:
Shouldn't this be higher by Solidity style guidelines (externals go above internals IIRC)? Might apply in other contracts as well.

Reply:
I enabled ordering rule in solhint and fixed it in whole repo.

I had to fork solhint since it currently does not support CustomErrors. I also created following PR
https://github.com/protofire/solhint/pull/377
Once merged (it is already approved) and released I will move Solhint back to official version